homeassistant/components/guardian/switch.py

```diff
@@ -12,10 +20,42 @@
 ATTR_INST_CURRENT_DDT = "instantaneous_current_ddt"
 ATTR_TRAVEL_COUNT = "travel_count"
 
+SERVICE_DISABLE_AP = "disable_ap"
+SERVICE_ENABLE_AP = "enable_ap"
+SERVICE_REBOOT = "reboot"
+SERVICE_RESET_VALVE_DIAGNOSTICS = "reset_valve_diagnostics"
+SERVICE_UPGRADE_FIRMWARE = "upgrade_firmware"
+
+SERVICE_UPGRADE_FIRMWARE_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_URL, default=DEFAULT_FIRMWARE_UPGRADE_URL): cv.url,
+        vol.Optional(CONF_PORT, default=DEFAULT_FIRMWARE_UPGRADE_PORT): cv.port,
+        vol.Optional(
+            CONF_FILENAME, default=DEFAULT_FIRMWARE_UPGRADE_FILENAME
+        ): cv.string,
+    }
+)
+
 
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up Guardian switches based on a config entry."""
     guardian = hass.data[DOMAIN][DATA_CLIENT][entry.entry_id]
+
+    platform = entity_platform.current_platform.get()
+
+    for service_name, schema, method in [
+        (SERVICE_DISABLE_AP, None, "async_disable_ap"),
```

Comment:
How will we target the entities without entity_id or area_id in the service call data (service schema)?

Reply:
Oh you're right. Entity Platform doesn't handle this case. I guess it's broken now 🤷